### PR TITLE
Add manual line break support.

### LINKED ISSUE #19: Add support for line breaks.
Djot supports [line breaks](https://djot.net/syntax#line-break), in a slightly different way that Markdown does: instead of double space, it uses `\` followed by a newline.

Since it is a frequent thing to use, it would be good to add support for those.

### PR DESCRIPTION
**Pull request by ghost**
This should close https://github.com/lpil/jot/issues/19. :)

### CODE REVIEW

File: test/cases/linebreak.test [hunk 1]
```diff
@@ -0,0 +1,9 @@
+```
+This is a soft
+break and this is a hard\
+break.
+.
+<p>This is a soft
+break and this is a hard<br>
+break.</p>
+```
```
Comment (lpil) [inline test/cases/linebreak.test]: Could you remove this and replace it with the canonical cases for this feature in `cases_unimplemented` please 🙏

**Review by lpil — COMMENTED** [review]
Nice! Thank you. Two small notes inline

File: test/cases/linebreak.test (hunk 1 above)
Comment (BConquest) [reply test/cases/linebreak.test:9]: Oh I was looking for the test case but my grep skills were failing me. Will take another look for it.

Comment (ghost) [reply test/cases/linebreak.test]: Yeah can do, I was looking for the test case in unimplemented but my grep skills have failed me. Will give it another look.

**Comment by ghost** [comment]
Hey, I have updated this branch to handle more of the escape sequences, the only test cases I can not figure out right now are the leading/trailing whitespace for handling hardbreaking newlines. I am also not positive on how the ascii punctuation formatting is handled but that is how `gleam format` handled it.

**Review by lpil — APPROVED** [review]
Thank you!
